pkg/cmdlets/errors/errors.go

```diff
@@ -85,6 +85,21 @@ func trimError(msg string) string {
 	return strings.TrimRight(msg, " .")
 }
 
+func (o *OutputError) MarshalStructured(f output.Format) interface{} {
+	var errors []interface{}
+	rerrs := locale.UnwrapError(o.error)
+	if len(rerrs) == 0 {
+		// It's possible the error came from cobra or something else low level that doesn't use localization
+		logging.Warning("Error does not have localization: %s", errs.JoinMessage(o.error))
+		rerrs = []error{o.error}
+	}
+	for _, errv := range rerrs {
+		message := trimError(locale.ErrorMessage(errv))
+		errors = append(errors, message)
+	}
+	return errors
+}
```

Comment:
Picking up from the discussion here:

https://github.com/ActiveState/cli/pull/2506#discussion_r1179664735

Can we make this return a slice of errors instead? Seems if it's only to satisfy the commented code linked above then the act of turning errors into strings is better handled when the error itself is reported.

Reply:
Not quite. I changed it to return a JSON error object directly (with a list of error strings) and changed the JSON `Error()` method to handle this case. It's possible to call JSON's `Error()` method directly from code (not just an error handler), so we still need to consider at least incoming `error`, `[]byte`, and `string` types.